synedrion/src/paillier/ring_pedersen.rs

```diff
@@ -72,36 +73,44 @@ impl<P: PaillierParams> RPParamsMod<P> {
     // - this will match the order in the paper
     pub fn commit(
         &self,
-        secret: &Signed<P::Uint>,
+        secret: &SecretBox<Signed<P::Uint>>,
         randomizer: &Signed<P::WideUint>,
     ) -> RPCommitmentMod<P> {
         // $t^\rho * s^m mod N$ where $\rho$ is the randomizer and $m$ is the secret.
         RPCommitmentMod(
-            pow_signed_wide::<P::Uint>(self.base, randomizer) * pow_signed(self.power, secret),
+            pow_signed_wide::<P::Uint>(self.base, randomizer)
+                * pow_signed(self.power, secret.expose_secret()),
         )
     }
 
     pub fn commit_wide(
         &self,
-        secret: &Signed<P::WideUint>,
+        // TODO(dp): @reviewers Question unrelated to the PR, just something I noticed: Why is the
```

Comment:
I guess it's just a naming issue. Technically, this method is `commit_wide_secret_wide_randomizer()`, and the next one is `commit_normal_secret_xwide_randomizer()`. These are just overrides for the same operations with different types of arguments. This could be made more straightforward if #60 is fixed.